test/integration/static-and-dynamic/controls/static-and-dynamic.rb

```diff
@@ -0,0 +1,65 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Roles amount are used to test how the module behaves on configuration updates.
+# Workaround InSpec lack of support for integer by parsing it from string.
+roles = attribute('roles').to_i
+
+project_groups = [
+  # Resource pairs (arrays of length = 2)
+  attribute('authoritative_static_projects'),
+  attribute('additive_static_projects'),
+  attribute('authoritative_dynamic_projects'),
+  attribute('additive_dynamic_projects')
+]
+
+# Member groupings
+member_groups = [
+  attribute('member_group_0'),
+  attribute('member_group_1')
+]
+
+# Projects
+
+for projects in project_groups do
```

Comment:
Again, we should avoid dynamically defining tests. It is acceptable to have repeated code in tests, and we can look at how to refactor it with a library if it becomes difficult to maintain.